File: python3/solutions/speedlimit.py
# Problem: https://open.kattis.com/problems/speedlimit

########## GENERATE TEST CASES LOCALLY ##########

dummy_input = [
  '3',
  '20 2',
  '30 6',
  '10 7',
  '2',
  '60 1',
  '30 5',
  '4',
  '15 1',
  '25 2',
  '30 3',
  '10 5',
  '-1'
]

def input():
  return dummy_input.pop(0)

################## CODE #####################

line = input()
data_set = []
result = ''

def compute_miles(data_set):
  sum_of_prev_time_values = 0
  total_miles = 0

  for value in data_set:
    spped_and_time = value.split(' ')
    speed = int(spped_and_time.pop(0))
    time = int(spped_and_time.pop(0)) - sum_of_prev_time_values
    sum_of_prev_time_values += time
    total_miles += speed * time
  return total_miles

while line != '-1':
  line = input()
  if line.find(' ') == -1:
    if len(data_set) > 0:
      result += str(compute_miles(data_set)) + ' miles\n'
    else:
      result += ''
    data_set = []
  else:
    data_set.append(line)


print(result)

File: javascript/solutions/speedlimit.js
// Problem: https://open.kattis.com/problems/speedlimit

let line;
let indexOfNewSet = 0;
let index = 0;
let result = '';
let dataSet = [];

while ((line = readline())) {
  if (index === indexOfNewSet) {
    // we've reached a data set separator ...
    // compute index of next separator
    indexOfNewSet += parseInt(line) + 1;
    // compute miles in prev data set
    result += dataSet.length > 0 ? computeMiles(dataSet) + ' miles\n' : '';
    dataSet = [];
  } else {
    // data set item, push data to set
    dataSet.push(line);
  }
  index += 1;
}

function computeMiles(dataSet) {
  let sumOfPrevTimeValues = 0;
  let totalMiles = 0;
  dataSet.forEach(value => {
    const sppedAndTime = value.split(' ');
    const speed = parseInt(sppedAndTime[0]);
    const time = parseInt(sppedAndTime[1]) - sumOfPrevTimeValues;
    sumOfPrevTimeValues += time;
    totalMiles += speed * time;
  });
  
  return totalMiles;
}

print(result);
